### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by alexeuler**
I added Shakacode linters + reworked all the code so it will actually pass all checks. One lint is still not resolved - as I understood this is a WIP component (Filtertabs).

---

This change is [Reviewable](https://reviewable.io/reviews/shakacode/todos-react-native-redux/2)

### CODE REVIEW

**Comment by justin808** [comment]
Looks great @alleycat-at-git.

@urbanvikingr Please take a peek.

 

---

Reviewed 41 of 41 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[.eslintrc.json, line 26 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8n5eEfcv5YcPfz2q0:-KM8n5eFVmW-s8FNALJ_:397691836) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/.eslintrc.json#L26)):*
> ```JSON
>   },
> 
>   "rules": {
> ```

We'll later want to consider moving these into a NPM package.

Any commentary on these choices? Are these the defaults?

---

*[app/containers/AddTodo.js, line 46 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8nrj9aaf6jqAfhV7m:-KM8nrj9aaf6jqAfhV7n:1378573077) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L46)):*
> ```JavaScript
> const AddTodo = connect()(AddTodoClass);
> 
> export default AddTodo;
> ```

Do we need a variable for the export default?

```
export default connect()(AddTodoClass);
```

Even if that works, I think that way that @alleycat-at-git changed the code is nice and clean.

---

*[app/containers/App.js, line 27 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8oGD-7AAwuqfWhedx:-KM8oGD0qXhSBvfWPS1_:-2016798670) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/App.js#L27)):*
> ```JavaScript
> );
> 
> export default App;
> ```

@robwise do we need the const var? or just export default?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-:-KM8ozni3yBnW5DVeL2R:-87486177)*

**Comment by alleycat-at-git** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[.eslintrc.json, line 26 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8n5eEfcv5YcPfz2q0:-KMGbFa6awRy5ZeA_keg:-1204643659) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/.eslintrc.json#L26)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> We'll later want to consider moving these into a NPM package.
> 
> Any commentary on these choices? Are these the defaults?
</details>
Yeah, these are defaults for react-native

---

*[app/containers/AddTodo.js, line 46 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8nrj9aaf6jqAfhV7m:-KMGbYMp4nWlKDXRV8mE:-570182194) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L46)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> Do we need a variable for the export default?
> 
> ```
> export default connect()(AddTodoClass);
> ```
> 
> Even if that works, I think that way that @alleycat-at-git changed the code is nice and clean.
> 
</details>
I think we don't, this is legal according to https://developer.mozilla.org/en/docs/web/javascript/reference/statements/export

---

*[app/containers/App.js, line 27 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8oGD-7AAwuqfWhedx:-KMGcJOb_C7gJmldkeme:-1780499653) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/App.js#L27)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> @robwise do we need the const var? or just export default?
</details>
We can't make just exports default here, because the App is yet not defined when we write 
```
export default App = ...
```

It works though for classes:
```
export default class App extends Component {...} 
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2)*

**Comment by justin808** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[app/containers/AddTodo.js, line 46 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8nrj9aaf6jqAfhV7m:-KMGlVH5JOjs1CbTa89b:613183010) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L46)):*
<details><summary><i>Previously, alleycat-at-git (Alexey) wrote…</i></summary>
> I think we don't, this is legal according to https://developer.mozilla.org/en/docs/web/javascript/reference/statements/export
</details>
Given the below as legal, should we remove the extra variable, @robwise, @alexfedoseev  

```js
export default expression;
export default function (…) { … } // also class, function*
export default function name1(…) { … } // also class, function*
export { name1 as default, … };
```

---

*[app/containers/App.js, line 27 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8oGD-7AAwuqfWhedx:-KMGlnFVkYiSB3FO8tB_:-1617565922) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/App.js#L27)):*
<details><summary><i>Previously, alleycat-at-git (Alexey) wrote…</i></summary>
> We can't make just exports default here, because the App is yet not defined when we write 
> ```
> export default App = ...
> ```
> 
> It works though for classes:
> ```
> export default class App extends Component {...} 
> ```
</details>
@alleycat-at-git , Why do you need the `App` reference?

```js

export default () => (
  <Provider store={store}>
    <Main />
  </Provider>
);
```

CC: @robwise 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2)*

File: app/components/TodoList.js [hunk 1]
```diff
@@ -1,67 +1,64 @@
...
-    })
+      dataSource: this.state.dataSource.cloneWithRowsAndSections(dataBlob, sectionIDs, rowIDs),
+    });
   }
   render() {
     return (
       <ListView
         automaticallyAdjustContentInsets={false}
-        enableEmptySections={true}
+        enableEmptySections
```
Comment (urbanvikingr) [inline app/components/TodoList.js]: Does this work without `{true}`?

File: app/components/TodoList.js [hunk 2]
```diff
@@ -70,45 +67,14 @@ export default class TodoList extends Component {
...
-    flexDirection: 'row',
-    alignItems: 'center'
-  },
-  sectionHeaderText: {
-    color: 'rgba(255,255,255,0.7)',
-    fontSize: 16
-  },
-  row: {
-    height: 48,
-    paddingHorizontal: 16,
```
Comment (urbanvikingr) [inline app/components/TodoList.js]: We need to be sure that we want to have styles outside component. There is a significant difference between RN and React. This isn't CSS as known from web. RN handles styles differently.

File: app/containers/App.js [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
-let createStoreWithMiddleware
+let createStoreWithMiddleware;
 
 if (__DEV__) {
-  createStoreWithMiddleware = applyMiddleware(thunk, logger)(createStore)
+  createStoreWithMiddleware = applyMiddleware(thunk, logger)(createStore);
 } else {
-  createStoreWithMiddleware = applyMiddleware(thunk)(createStore)
+  createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
 }
```
Comment (urbanvikingr) [inline app/containers/App.js]: Let's more this into a file `store.js`. What do you think @alleycat-at-git @robwise ?

**Comment by robwise** [comment]
Reviewed 38 of 41 files at r1.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[package.json, line 26 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHojYpDGcIS-QUqtiJ:-KMHojYqvnqh-80FZvEi:-2026232514) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/package.json#L26)):*
> ```JSON
>     "mocha": "^2.4.5",
>     "react-addons-test-utils": "0.14.8",
>     "react-dom": "0.14.8",
> ```

`react-dom` is dev-only?

---

*[app/actions/index.js, line 5 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHozDaUA_U0-8kk2a4:-KMHozDaUA_U0-8kk2a5:-900134738) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/actions/index.js#L5)):*
> ```JavaScript
> 
> let nextTodoId = 0;
> export const addTodo = text => (
> ```

why not use `makeAction` here?

---

*[app/components/Main.js, line 28 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHp792RpxDLauCpf_-:-KMHp792RpxDLauCpf_0:-1582879485) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/components/Main.js#L28)):*
> ```JavaScript
>   configureScene() {
>     return Navigator.SceneConfigs.FloatFromBottom;
>   }
> ```

looks like all three of these methods can be inlined, and therefore this entire thing can become an SFC, which is preferable IMO

---

*[app/components/TodoList.js, line 48 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8mtMlD2Sd4Xah7n1g-r1-48:-KMHmEeOjCiiTu08KcC6:160424895) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/components/TodoList.js#L48)):*
<details><summary><i>Previously, urbanvikingr (Dan Jensen) wrote…</i></summary>
> Does this work without `{true}`? @alleycat-at-git 
</details>
yep

---

*[app/components/Todos.js, line 35 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHpXqtJ_2ekW-yEz6p:-KMHpXqtJ_2ekW-yEz6q:-1508066678) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/components/Todos.js#L35)):*
> ```JavaScript
>     );
>   }
>   change(state: {}) {
> ```

is this a flow annotation or something? You're not really using flow annotations elsewhere in your components as far as I can tell. It may be advantageous to keep it consistent. Also, same comment here about the `change` method, I don't really get its purpose

---

*[app/containers/AddTodo.js, line 6 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHmN-M-UWKqa_W9tb0:-KMHmN-M-UWKqa_W9tb1:1305527082) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L6)):*
> ```JavaScript
> import styles from '../styles/addTodo';
> 
> class AddTodoClass extends Component {
> ```

I would name this `AddTodoContainer` not `Class`?

---

*[app/containers/AddTodo.js, line 36 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHnUfhro3uoT8FDPQN:-KMHnUfhro3uoT8FDPQO:-1690549152) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L36)):*
> ```JavaScript
>   }
>   onSubmit() {
>     this.props.dispatch(addTodo(this.state.text));
> ```

normally we would bind props to dispatch as the second argument to the `connect` function so we can avoid having to do this

---

*[app/containers/AddTodo.js, line 39 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHnYwANVW83rI_XHPA:-KMHnYwBc65PjnmBnd-G:1276776853) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L39)):*
> ```JavaScript
>     this.change({ text: '' });
>   }
>   change(state) {
> ```

I would delete this method

---

*[app/containers/AddTodo.js, line 46 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8nrj9aaf6jqAfhV7m:-KMHmWnvv9owAQyMm1sf:-125236658) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L46)):*
> Do we need a variable for the export default?

No you don't, you can just export it directly since it doesn't need to be named.

So Alex and I would tend to just do it in one line:

```js
export default connect()(AddTodoClass);
```

Also, anytime you are adding `Class` or `Value` or just anything that really means exactly the same thing as what the variable originally meant, only to avoid a naming collision, is usually a smell that you don't need to be defining both variables, or if you are, you should name them with something that indicates what is different about them.

---

*[app/containers/App.js, line 17 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8mtMmyLZEMjehrJVU-r1-17:-KMHoTKpmgjXmVABSRRw:348655486) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/App.js#L17)):*
<details><summary><i>Previously, urbanvikingr (Dan Jensen) wrote…</i></summary>
> Let's more this into a file `store.js`. What do you think @alleycat-at-git @robwise ?
</details>
yeah that's what we tend to do with F&G, and we import the root reducer there as well

---

*[app/containers/App.js, line 27 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8oGD-7AAwuqfWhedx:-KMHoBOznSDF7X3xRF4X:-19833727) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/App.js#L27)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> @alleycat-at-git , Why do you need the `App` reference?
> 
> ```js
> 
> export default () => (
>   <Provider store={store}>
>     <Main />
>   </Provider>
> );
> ```
> 
> CC: @robwise 
</details>
I'm pretty sure Justin's way would work fine

---

*[app/utils/getDataSource.js, line 1 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHrzKWOX2XfkclA5al:-KMHrzKXgipvvzplU4Jh:1582776991) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/utils/getDataSource.js#L1)):*
> ```JavaScript
> export const getDataSource = data => {
> ```

if you're going to be using flow in this project, this is one place that _badly_ needs it. Clearly, there is some sort of complex nested structure to `data` that this entire function relies on being "just so." It's hard to understand what's going on without some type of description in this file of what this file looks like, which is what a Flow annotation would provide.

Also, this code is highly imperative. I would suggest using lodash/fp to switch to a more declarative style. Lastly, this data structure seems to be highly redundant.

---

*[app/utils/sortTodos.js, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHvLc9uTAXD5wRElHE:-KMHvLc9uTAXD5wRElHF:-2140828103) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/utils/sortTodos.js#L14)):*
> ```JavaScript
> );
> 
> const cumulateTodos = (todos: [], todo: {}) => (
> ```

technically `cumulate` is a word, though usually we use `accumulate`, is that what you mean here?

---

*[test/actions/actions.spec.js, line 16 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHvncmHtnmwfxUN1DE:-KMHvncmHtnmwfxUN1DF:-395121785) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/test/actions/actions.spec.js#L16)):*
> ```JavaScript
>     );
>   });
>   it('setVisibilityFilter should create SET_VISIBILITY_FILTER action', () => {
> ```

according to mocha docs they frown upon fat arrows for some reason, some sort of weird edge case, I still do it in my code but just throwing it out there in case you experience some weird bug: https://mochajs.org/#arrow-functions

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2)*

**Comment by robwise** [comment]
Reviewed 3 of 41 files at r1.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2)*

**Comment by justin808** [comment]
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[app/containers/AddTodo.js, line 6 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHmN-M-UWKqa_W9tb0:-KMJ4RkzxOUKT5TSQiwi:262468467) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L6)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> I would name this `AddTodoContainer` not `Class`?
</details>
:thumbsup: 

---

*[app/containers/AddTodo.js, line 36 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHnUfhro3uoT8FDPQN:-KMJ4ZAjwjCDzqmN-0U_:-1662725392) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L36)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> normally we would bind props to dispatch as the second argument to the `connect` function so we can avoid having to do this
</details>
Not sure what you mean by that @robwise 

---

*[app/containers/App.js, line 17 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KM8mtMmyLZEMjehrJVU-r1-17:-KMJ4nbKB5XWEaDfZJkH:262468467) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/App.js#L17)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> yeah that's what we tend to do with F&G, and we import the root reducer there as well
</details>
:thumbsup: 

---

*[app/utils/getDataSource.js, line 1 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHrzKWOX2XfkclA5al:-KMJ4rnZrIeuZEHVCBIw:262468467) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/utils/getDataSource.js#L1)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> if you're going to be using flow in this project, this is one place that _badly_ needs it. Clearly, there is some sort of complex nested structure to `data` that this entire function relies on being "just so." It's hard to understand what's going on without some type of description in this file of what this file looks like, which is what a Flow annotation would provide.
> 
> Also, this code is highly imperative. I would suggest using lodash/fp to switch to a more declarative style. Lastly, this data structure seems to be highly redundant.
</details>
:thumbsup: 

---

*[app/utils/sortTodos.js, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHvLc9uTAXD5wRElHE:-KMJ4wL1JXxD5lZkQ7dr:262468467) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/utils/sortTodos.js#L14)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> technically `cumulate` is a word, though usually we use `accumulate`, is that what you mean here?
</details>
:thumbsup: 

---

*[test/actions/actions.spec.js, line 16 \[r1\]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHvncmHtnmwfxUN1DE:-KMJ55Oo8bJC2hQps_P4:1186353542) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/test/actions/actions.spec.js#L16)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> according to mocha docs they frown upon fat arrows for some reason, some sort of weird edge case, I still do it in my code but just throwing it out there in case you experience some weird bug: https://mochajs.org/#arrow-functions
</details>
the reason is legit in that function() { } is just as well and does the correct thing of not binding the `this`. I'd stick to the docs.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2)*

**Comment by robwise** [comment]
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

_[app/containers/AddTodo.js, line 36 [r1]](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2#-KMHnUfhro3uoT8FDPQN:-KMLlPR6iHvAXJGD4jol:144414809) ([raw file](https://github.com/shakacode/todos-react-native-redux/blob/3ea77bd84fd89e4786118f2ed5613733b262c93e/app/containers/AddTodo.js#L36)):_
<details><summary>

<i>Previously, justin808 (Justin Gordon) wrote…</i></summary>



> Not sure what you mean by that @robwise 
> </details>
> @justin808  I meant actions not props, but this is straight out of the README: http://redux.js.org/docs/api/bindActionCreators.html

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/todos-react-native-redux/2)_
